app/src/main/java/com/keylesspalace/tusky/components/timeline/viewmodel/CachedTimelineViewModel.kt

```diff
@@ -0,0 +1,187 @@
+import androidx.paging.ExperimentalPagingApi
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.cachedIn
+import androidx.paging.filter
+import androidx.paging.map
+import androidx.room.withTransaction
+import com.google.gson.Gson
+import com.keylesspalace.tusky.appstore.BookmarkEvent
+import com.keylesspalace.tusky.appstore.EventHub
+import com.keylesspalace.tusky.appstore.FavoriteEvent
+import com.keylesspalace.tusky.appstore.PinEvent
+import com.keylesspalace.tusky.appstore.ReblogEvent
+import com.keylesspalace.tusky.components.timeline.Placeholder
+import com.keylesspalace.tusky.components.timeline.toEntity
+import com.keylesspalace.tusky.components.timeline.toViewData
+import com.keylesspalace.tusky.db.AccountManager
+import com.keylesspalace.tusky.db.AppDatabase
+import com.keylesspalace.tusky.entity.Poll
+import com.keylesspalace.tusky.network.FilterModel
+import com.keylesspalace.tusky.network.MastodonApi
+import com.keylesspalace.tusky.network.TimelineCases
+import com.keylesspalace.tusky.util.dec
+import com.keylesspalace.tusky.util.inc
+import com.keylesspalace.tusky.viewdata.StatusViewData
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.rx3.await
+import javax.inject.Inject
+
+class CachedTimelineViewModel @Inject constructor(
+    timelineCases: TimelineCases,
+    private val api: MastodonApi,
+    eventHub: EventHub,
+    accountManager: AccountManager,
+    sharedPreferences: SharedPreferences,
+    filterModel: FilterModel,
+    private val db: AppDatabase,
+    private val gson: Gson
+) : TimelineViewModel(timelineCases, api, eventHub, accountManager, sharedPreferences, filterModel) {
+
+    @ExperimentalPagingApi
+    override val statuses = Pager(
+        config = PagingConfig(pageSize = LOAD_AT_ONCE),
+        remoteMediator = CachedTimelineRemoteMediator(accountManager, api, db, gson),
+        pagingSourceFactory = { db.timelineDao().getStatusesForAccount(accountManager.activeAccount!!.id) }
+    ).flow
+        .map { pagingData ->
+            pagingData.map { timelineStatus ->
+                timelineStatus.toViewData(gson)
+            }
+        }
+        .map { pagingData ->
+            pagingData.filter { statusViewData ->
+                !shouldFilterStatus(statusViewData)
+            }
+        }
+        .cachedIn(viewModelScope)
+
+    override fun updatePoll(newPoll: Poll, status: StatusViewData.Concrete) {
+        // handled by CacheUpdater
+    }
+
+    override fun changeExpanded(expanded: Boolean, status: StatusViewData.Concrete) {
+        viewModelScope.launch {
+            db.timelineDao().setExpanded(accountManager.activeAccount!!.id, status.id, expanded)
+        }
+    }
+
+    override fun changeContentHidden(isShowing: Boolean, status: StatusViewData.Concrete) {
+        viewModelScope.launch {
+            db.timelineDao().setContentHidden(accountManager.activeAccount!!.id, status.id, isShowing)
```

Comment:
```suggestion
            db.timelineDao().setContentHidden(accountManager.activeAccount!!.id, status.id, !isShowing)
```

Reply:
No, that should be fine the way it is

Reply:
oh, no, because of the inverted logic 🤯 
Let me fix this properly by using the same name everywhere, this is too confusing